apps/prairielearn/src/pages/instructorAssessmentInstances/instructorAssessmentInstances.js

```diff
@@ -1,191 +1,204 @@
-    checkBelongs(assessment_instance_id, assessment_id, (err) => {
-      if (ERR(err, next)) return;
-
+      res.send(JSON.stringify({}));
+    } else if (req.body.__action === 'delete') {
+      const assessment_id = res.locals.assessment.id;
+      const assessment_instance_id = req.body.assessment_instance_id;
+      checkBelongsAsync(assessment_instance_id, assessment_id);
       const params = [assessment_instance_id, res.locals.authn_user.user_id];
-      sqldb.call('assessment_instances_delete', params, function (err) {
-        if (ERR(err, next)) return;
-        res.send(JSON.stringify({}));
-      });
-    });
-  } else if (req.body.__action === 'grade_all' || req.body.__action === 'close_all') {
-    const assessment_id = res.locals.assessment.id;
-    const close = req.body.__action === 'close_all';
-    const overrideGradeRate = true;
-    util.callbackify(gradeAllAssessmentInstances)(
-      assessment_id,
-      res.locals.user.user_id,
-      res.locals.authn_user.user_id,
-      close,
-      overrideGradeRate,
-      function (err, job_sequence_id) {
-        if (ERR(err, next)) return;
-        res.redirect(res.locals.urlPrefix + '/jobSequence/' + job_sequence_id);
-      },
-    );
-  } else if (req.body.__action === 'delete_all') {
-    const params = [res.locals.assessment.id, res.locals.authn_user.user_id];
-    sqldb.call('assessment_instances_delete_all', params, function (err) {
-      if (ERR(err, next)) return;
+      sqldb.callAsync('assessment_instances_delete', params);
       res.send(JSON.stringify({}));
-    });
-  } else if (req.body.__action === 'regrade') {
-    const assessment_id = res.locals.assessment.id;
-    const assessment_instance_id = req.body.assessment_instance_id;
-    checkBelongs(assessment_instance_id, assessment_id, (err) => {
-      if (ERR(err, next)) return;
-
-      regradeAssessmentInstance(
+    } else if (req.body.__action === 'grade_all' || req.body.__action === 'close_all') {
+      const assessment_id = res.locals.assessment.id;
+      const close = req.body.__action === 'close_all';
+      const overrideGradeRate = true;
+      const job_sequence_id = await gradeAllAssessmentInstances(
+        assessment_id,
+        res.locals.user.user_id,
+        res.locals.authn_user.user_id,
+        close,
+        overrideGradeRate,
+      );
+      res.redirect(res.locals.urlPrefix + '/jobSequence/' + job_sequence_id);
+    } else if (req.body.__action === 'delete_all') {
+      const params = [res.locals.assessment.id, res.locals.authn_user.user_id];
+      sqldb.callAsync('assessment_instances_delete_all', params);
+      res.send(JSON.stringify({}));
+    } else if (req.body.__action === 'regrade') {
+      const assessment_id = res.locals.assessment.id;
+      const assessment_instance_id = req.body.assessment_instance_id;
+      checkBelongsAsync(assessment_instance_id, assessment_id);
+      const job_sequence_id = regradeAssessmentInstance(
```

Comment:
AFAICT this was never refactored into an async function, so it's not safe to test like this. Did you test the changes from this PR?

Reply:
`regradeAssessmentInstance` has now been refactored into an  async function in commit [8472ae8](https://github.com/PrairieLearn/PrairieLearn/pull/9080/commits/8472ae8a0869182f87c71e15b748cbd6064db927).

Reply:
I left a note about this refactor here: https://github.com/PrairieLearn/PrairieLearn/pull/9080#discussion_r1441085953. Let's continue discussion on that new thread.